### DISCUSSION TITLE
Is it possible to fetch the default language of a channel with the innertube API?

### DISCUSSION THREAD

**Discussion by TasseDeCafe** [LuanRT/YouTube.js#488] [General]
In YouTube Studio, it is possible to set a default language for a channel.

I'm trying to figure out how YouTube selects the default language of the subtitles (the language that appears by default when you click on the gear button in the video player), but I just can't find it.

<img width="281" alt="image" src="https://github.com/LuanRT/YouTube.js/assets/44320355/a5571281-a4ae-428a-a3c1-be6190e371ed">

I assumed that this information would be loaded by YouTube through the innertube API, but I can't find it in the dev tools or by using the getChannel method of this library.

Do you you know how to find this piece of information?

Thank you!

**Comment by absidue** [comment]
You can't get the default channel language unless you are the owner of the channel, but you can figure out which captions track YouTube picks by default for a given video, by looking captions property in the video info returned by `Innertube.getBasicInfo` and `Innertube.getInfo`.

**Comment by TasseDeCafe** [comment]
Hi! Thank you for your help. Unfortunately many creators seem to pick a default caption language that is different than the language spoken in the video. I'm assuming that this is because they want their viewers to be able to understand them.

This video for example: 

https://www.youtube.com/watch?v=ID7VkY6cMRQ

will default to English for the captions, but the language spoken is Russian. It must be possible to get the language code of the language spoken because this extension does it: https://chrome.google.com/webstore/detail/language-reactor/hoombieeljmmljlkjmnheibnpciblicm

But I really can't figure out how they do it. The captions property doesn't contain this piece of data...

**Reply by absidue** [reply]
> The captions property doesn't contain this piece of data...

Yes it does :). You need to look for the auto-generated captions track (you can check if the `kind` property is `asr`) as that will always be in the language of the video. For videos that don't have an auto-generated captions track, you can't get the language of the video from an API response.

**Reply by TasseDeCafe** [reply]
Indeed! You're right, I didn't notice that. Thank you!
